options.go

```diff
@@ -174,21 +172,6 @@ func (o *PullOptions) Validate() error {
 	return nil
 }
 
-type TagMode int
-
-const (
-	InvalidTagMode TagMode = iota
-	// TagFollowing any tag that points into the histories being fetched is also
-	// fetched. TagFollowing requires a server with `include-tag` capability
-	// in order to fetch the annotated tags objects.
-	TagFollowing
-	// AllTags fetch all tags from the remote (i.e., fetch remote tags
-	// refs/tags/* into local tags with the same name)
-	AllTags
-	// NoTags fetch no tags from the remote at all
-	NoTags
-)
```

Comment:
As we are still early days for the V6 release, it may be beneficial to keep a type redirect so that users consuming `v6-exp` (soon to become `main`), don't have to change their code.

```golang
// TODO: delete for V6
type TagMode = plumbing.TagMode

const (
	InvalidTagMode = plumbing.InvalidTagMode
	TagFollowing   = plumbing.TagFollowing
	AllTags        = plumbing.AllTags
	NoTags         = plumbing.NoTags
)
```

Reply:
Fair enough. Updated the patch